books/forms.py

```diff
@@ -53,7 +53,10 @@ def clean_cover(self):
 
     def clean_isbn(self):
         isbn = self.cleaned_data.get("isbn")
-        return "".join(isbn.split("-"))
+        if isbn:
+            return "".join(isbn.split("-"))
+        else:
+            return None
```

Comment:
A common way to write the `if/else` for small things would be:
```
return isbn and "".join(isbn.split("-"))
```

But also, instead of spliting and re-joining, you can just replace dashes with... nothing:

```
return isbn and isbn.replace("-", "")
```

And if you want to also clear any whitespace, you could use a simple regex to clear everything:

```
import re
seperators = re.compile("[\-\s]") # \s stands for whitespace characters
return isbn and seperators.sub("", isbn)
```